contracts/transfer/circuits/src/gadgets/preimage.rs

```diff
@@ -4,37 +4,52 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
+use dusk_plonk::bls12_381::Scalar as BlsScalar;
+use dusk_plonk::constraint_system::ecc::Point as PlonkPoint;
+use dusk_plonk::jubjub::AffinePoint;
 use dusk_plonk::prelude::*;
 use plonk_gadgets::AllocatedScalar;
-use poseidon252::sponge::sponge::sponge_hash_gadget;
+use poseidon252::sponge::sponge::{sponge_hash, sponge_hash_gadget};
+use std::convert::TryInto;
 
 /// Prove knowledge of the preimage of a note,
 /// used as input for a transaction.
 #[allow(non_snake_case)]
 pub fn input_preimage(
     composer: &mut StandardComposer,
-    value_commitment_x: AllocatedScalar,
-    value_commitment_y: AllocatedScalar,
-    pos: AllocatedScalar,
-    pk_r_x: AllocatedScalar,
-    pk_r_y: AllocatedScalar,
+    note_type: AllocatedScalar,
+    value_commitment: PlonkPoint,
+    nonce: AllocatedScalar,
+    pk_r: PlonkPoint,
+    randomness: PlonkPoint,
+    position: AllocatedScalar,
+    cipher_one: AllocatedScalar,
+    cipher_two: AllocatedScalar,
+    cipher_three: AllocatedScalar,
 ) -> Variable {
     let output = sponge_hash_gadget(
```

Comment:
No need to assingn the result to a variable and return it. Just return the result of the function without allocating it.